tests/wallet/sync/test_wallet_sync.py

```diff
@@ -23,6 +27,97 @@ def wallet_height_at_least(wallet_node, h):
+        ph = await wallet.get_new_puzzlehash()
+        for block in default_1000_blocks[:100]:
+            await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(block))
+
+        msg = await full_node_api.request_block_headers(
+            wallet_protocol.RequestBlockHeaders(uint32(10), uint32(15), False)
+        )
+        assert msg.type == ProtocolMessageTypes.respond_block_headers.value
+        res_block_headers = RespondBlockHeaders.from_bytes(msg.data)
+        bh = res_block_headers.header_blocks
+        assert len(bh) == 6
+        assert [x.reward_chain_block.height for x in default_1000_blocks[10:16]] == [
+            x.reward_chain_block.height for x in bh
+        ]
+
+        assert [x.foliage for x in default_1000_blocks[10:16]] == [x.foliage for x in bh]
+
+        assert [x.transactions_filter for x in bh] == [b"\x00"] * 6
+
+        num_blocks = 20
+        new_blocks = bt.get_consecutive_blocks(
+            num_blocks, block_list_input=default_1000_blocks, pool_reward_puzzle_hash=ph
+        )
+        for i in range(0, len(new_blocks)):
+            await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(new_blocks[i]))
+
+        msg = await full_node_api.request_block_headers(
+            wallet_protocol.RequestBlockHeaders(uint32(110), uint32(115), True)
+        )
+        res_block_headers = RespondBlockHeaders.from_bytes(msg.data)
+        bh = res_block_headers.header_blocks
+        assert len(bh) == 6
+
+    # @pytest.mark.parametrize(
+    #     "test_case",
+    #     [(1000000, 10000010, False, ProtocolMessageTypes.reject_block_headers)],
+    #     [(80, 99, False, ProtocolMessageTypes.respond_block_headers)],
+    #     [(10, 8, False, None)],
+    # )
+    @pytest.mark.asyncio
+    async def test_request_block_headers_rejected(self, bt, wallet_node, default_1000_blocks):
+        # Tests the edge case of receiving funds right before the recent blocks  in weight proof
+        full_node_api: FullNodeAPI
+        full_node_api, wallet_node, full_node_server, wallet_server = wallet_node
+
+        # start_height, end_height, return_filter, expected_res = test_case
+
+        msg = await full_node_api.request_block_headers(
+            wallet_protocol.RequestBlockHeaders(uint32(1000000), uint32(1000010), False)
+        )
+        assert msg.type == ProtocolMessageTypes.reject_block_headers.value
+
+        for block in default_1000_blocks[:150]:
+            await full_node_api.full_node.respond_block(full_node_protocol.RespondBlock(block))
+
+        msg = await full_node_api.request_block_headers(
+            wallet_protocol.RequestBlockHeaders(uint32(80), uint32(99), False)
+        )
+        assert msg.type == ProtocolMessageTypes.respond_block_headers.value
+        msg = await full_node_api.request_block_headers(
+            wallet_protocol.RequestBlockHeaders(uint32(10), uint32(8), False)
+        )
+        assert msg is None
```

Comment:
is there a good reason for returning `None` instead of a rejection here?

Reply:
was trying to make it backwards compatible, but rejection would make more sense in this case, not sure why other calls return None. Rejection seems to be reserved for missing hashes only.